sampler.go

```diff
@@ -212,37 +252,44 @@ func NewRemotelyControlledSampler(
 
 // IsSampled implements IsSampled() of Sampler.
 func (s *RemotelyControlledSampler) IsSampled(id uint64) bool {
-	s.lock.RLock()
-	defer s.lock.RUnlock()
+	s.RLock()
+	defer s.RUnlock()
 	return s.sampler.IsSampled(id)
 }
 
 // Close implements Close() of Sampler.
 func (s *RemotelyControlledSampler) Close() {
-	s.lock.RLock()
+	s.RLock()
 	s.timer.Stop()
-	s.lock.RUnlock()
+	s.RUnlock()
 
 	s.pollStopped.Wait()
 }
 
+// Tags implements Tags of Sampler
+func (s *RemotelyControlledSampler) getTags() []tag {
+	s.RLock()
+	defer s.RUnlock()
+	return s.sampler.getTags()
+}
+
 // Equal implements Equal() of Sampler.
 func (s *RemotelyControlledSampler) Equal(other Sampler) bool {
 	if o, ok := other.(*RemotelyControlledSampler); ok {
-		s.lock.RLock()
-		o.lock.RLock()
-		defer s.lock.RUnlock()
-		defer o.lock.RUnlock()
+		s.RLock()
+		o.RLock()
+		defer s.RUnlock()
```

Comment:
* creating an anonymous function that captures the variables requires memory allocation
 * there is no difference in which order these to are unlocked